src/database/operations/users/loadUserByKeycloakUserId.ts

```diff
@@ -2,20 +2,20 @@ import { db } from '../../db';
 import { NotFoundError } from '../../../errors';
 import { JsonResultSet, type User } from '../../../types';
 
-export const loadUserByAuthenticationId = async (
-	authenticationId: string,
+export const loadUserByKeycloakUserId = async (
+	keycloakUserId: string,
```

Comment:
I'm disappointed the [`uuid` package](https://www.npmjs.com/package/uuid) (or its [types](https://www.npmjs.com/package/@types/uuid)) don't contain a `Uuid` type; is it worth making a newtype for this? `string` is so vague!

Reply:
Ha I actually did just start to do this when working on the next commit (and was going to backport it here).

This will be handy: https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html

Reply:
OK so I looked into the literal types and... it would be a bit of a bear / I'm not convinced it would actually be a reasonable type to make.

I did make a `KeycloakUserId` type but it's just an alias for `string` (which means it isn't providing any actual type safety beyond just being a string).

This means TypeScript would allow, for instance:

```
const notAUuid = `lol`

const user: User = {
  keycloakUserId: notAUuid,
}
```

Reply:
OH -- I'm seeing you said `newtype for this` not `new type for this`

and seeing that's what newtype is for.

Ok, sorry, I'm caught up now.  Looking into it all.

Reply:
I've implemented this as a separate commit; it's an augmentation and new pattern, and a fairly beefy one all considered so felt appropriate to implement it as a second step.